psi4/src/export_mints.cc

```diff
@@ -406,37 +415,44 @@ void export_mints(py::module& m)
         .def("symmetrize_gradient", &Matrix::symmetrize_gradient, "docstring")
         .def("rotate_columns", &Matrix::rotate_columns, "docstring")
         .def("array_interface", [](Matrix& m) {
-            // Dont ask, hopefully pybind11 will work on this
+
+            // Build a list of NumPy views, used for the .np and .nph accessors.Vy
             py::list ret;
 
             // If we set a NumPy shape
             if (m.numpy_shape().size()) {
-                if (m.nirrep() > 1){
-                    throw PSIEXCEPTION("Vector::array_interface numpy shape with more than one irrep is not valid.");
+                if (m.nirrep() > 1) {
+                    throw PSIEXCEPTION(
+                        "Vector::array_interface numpy shape with more than one irrep is not "
+                        "valid.");
+                }
+
+                // Cast the NumPy shape vector
+                std::vector<size_t> shape;
+                for (int val : m.numpy_shape()) {
+                    shape.push_back((size_t)val);
                 }
 
-                py::dict interface;
-                interface["typestr"] = py::format_descriptor<double>::format();
-                interface["data"] = py::make_tuple((long)(m.pointer(0)[0]), false);
-                interface["shape"] = m.numpy_shape();
-                ret.append(interface);
+                // Build the array
+                py::array arr(shape, m.pointer(0)[0], py::cast(&m));
+                ret.append(arr);
 
             } else {
                 for (size_t h = 0; h < m.nirrep(); h++) {
-                    py::dict interface;
-                    interface["typestr"] = py::format_descriptor<double>::format();
-                    if ((m.rowdim(h) == 0) || (m.coldim(h) == 0)) {
-                        interface["data"] = py::make_tuple(0, false);
-                    } else {
-                        interface["data"] = py::make_tuple((long)(m.pointer(h)[0]), false);
+                    // Hmm, sometimes we need to overload to nullptr
+                    double* ptr = nullptr;
```

Comment:
Nice use of nullptr. This is something else we should adopt in psi4 while passing out NULL or 0.